sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py

```diff
@@ -275,13 +279,19 @@ def add_content(self,
         """Delegate additional content creation.
 
         """
-
         super().add_content(more_content, **kwargs)
 
         # do not provide any additional info if documented as attribute
         if self.doc_as_attr:
             return
 
+        if self.pydantic.options.is_true("erdantic-figure", True):
+            if self.pydantic.options.is_true(
+                    "erdantic-figure-collapsed", True):
```

Comment:
In my opinion, we can move `self.pydantic.options.is_true("erdantic-figure-collapsed", True)` into the `self.add_erdantic_figure`. This would remove the `if` condition here and also would make the `collapsed` argument of `self.add_erdantic_figure` obsolete. Overall, that would be a minor reduction of complexity.

Reply:
fixed by ff24325